bullet_train/app/controllers/concerns/account/controllers/base.rb

```diff
@@ -64,7 +64,7 @@ def ensure_onboarding_is_complete_and_set_next_step
   def ensure_onboarding_is_complete
     # This is temporary, but if we've gotten this far and `@team` is loaded, we need to ensure current_team is
     # updated for the checks below. This entire concept of `current_team` is going away soon.
```

Comment:
> This entire concept of `current_team` is going away soon.

Interesting! @gazayas it looks like this was added 2 years ago, do you know what the plans were for this?

Reply:
I'd love to see `user.current_team` go away. That kind of thing has the potential to cause unexpected behavior if someone has multiple tabs open on the app and are jumping back and forth between them.

Reply:
Yeah, that's where I like a `Current` class to maintain that for us. We might not need to save `current_team` on the user, it's more of a transient attribute, right?

It looks like we've got a sorta setup `Current` object (though there's a bug in `update_membership`, `where` -> `find_by`) https://github.com/bullet-train-co/bullet_train-core/blob/3a46ebe4d6992908274e132833c0acfb6dacf40b/bullet_train/app/models/concerns/current_attributes/base.rb

Reply:
If we know what the behavior is that we want, I can look at updating/fixing `Current` since I've worked with those before.

Reply:
I honestly don't know enough about `user.current_team` and how/where/why we depend on it to be confident that I could describe what behavior we want in any more detail than "do the right thing", which... isn't very helpful.

Reply:
> > This entire concept of `current_team` is going away soon.
> 
> Interesting! @gazayas it looks like this was added 2 years ago, do you know what the plans were for this?

@kaspth To be 100% honest I'm not sure entirely what the plans for this were, but using `Current` to maintain this makes sense to me. If we do remove `current_team` I could see it having a lot of effects on this PR, but implementing that first and then revisiting this PR might be in our best interest.

Reply:
💪